Userland/xargs.cpp

```diff
@@ -0,0 +1,186 @@
+    size_t index = 0;
+    bool some_argument_has_replacement = false;
+
+    for (auto argument : arguments) {
+        StringView arg { argument };
+        Vector<StringView> parts;
+        size_t previous_index = 0;
+        if (auto first_index = arg.find_first_of(replace_view); first_index.has_value()) {
+            some_argument_has_replacement = true;
+            while (first_index.has_value()) {
+                auto current_index = first_index.value();
+                auto substring_until_replacement = arg.substring_view(previous_index, current_index);
+                parts.append(substring_until_replacement);
+                previous_index = current_index;
+                first_index = arg.substring_view(current_index + replace_view.length(), arg.length() - current_index - replace_view.length()).find_first_of(replace_view);
+            }
+            parts.append(arg.substring_view(previous_index + replace_view.length(), arg.length() - previous_index - replace_view.length()));
+        } else {
+            parts.append(arg);
+        }
+        argument_parts.append(move(parts));
+        ++index;
+    }
+
+    // Append a '{}' at the end if no arguments have replacements.
+    if (!some_argument_has_replacement) {
+        Vector<StringView> parts;
+        parts.append("");
+        parts.append("");
+        argument_parts.append(move(parts));
+    }
+
+    FILE* fp = stdin;
+    bool is_stdin = true;
+
+    if (StringView { "-" } != file_to_read) {
+        // A file was specified, try to open it.
+        fp = fopen(file_to_read, "r");
+        if (!fp) {
+            perror("fopen");
+            return 1;
+        }
+        is_stdin = false;
+    }
+
+    char* line = nullptr;
+    size_t buffer_size = 0;
+
+    StringBuilder builder;
+    Vector<char*> child_argv;
+
+    child_argv.resize(argument_parts.size() + 2);
+    child_argv[0] = const_cast<char*>(command);
+    child_argv[child_argv.size() - 1] = 0;
+
+    int devnull_fd = 0;
+
+    if (is_stdin) {
+        devnull_fd = open("/dev/null", O_RDONLY);
+        if (devnull_fd < 0) {
+            perror("open");
+            return 1;
+        }
+    }
+
+    for (;;) {
+        auto line_size = getdelim(&line, &buffer_size, entry_separator, fp);
+
+        if (line_size <= 0)
+            break;
+
+        if (line[line_size - 1] == '\n')
+            --line_size;
+
+        StringView line_view { line, (size_t)line_size };
+
+        size_t index = 0;
+        for (auto& parts : argument_parts) {
+            builder.clear();
+            builder.join(line_view, parts);
+            child_argv[index + 1] = strdup(builder.build().characters());
+            ++index;
+        }
+
+        if (verbose) {
+            fprintf(stderr, "xargs: ");
+            for (auto* ptr : child_argv) {
+                if (!ptr)
+                    break;
+                fprintf(stderr, "%s ", ptr);
+            }
+            fprintf(stderr, "\n");
```

Comment:
Flush here.

(Though in case of stderr specifically, buffering is disabled by default on Serenity.)

Reply:
stderr shouldn't be buffered anywhere...anyone that does that is insane